tribe-ext-schedule-day-view.php

```diff
@@ -0,0 +1,886 @@
+		 * without the "All Day" box checked (i.e. start and end times are set).
+		 *
+		 * If an event started a previous day (yesterday or prior) but is still
+		 * happening ("has not yet ended", nothing to do with recurrence):
+		 * A) Make it appear as an All Day event if its end time is beyond today
+		 * B) Or make its hour = -1 (not a real hour number), which will show up
+		 * in the Morning time slot
+		 * C) Else get the hour number (e.g. `23` for 11pm)
+		 * Most events will fall into the "C" scenario.
+		 * If an event falls into scenario "A" or "B", its WP_Post
+		 * `$post->timeslot` (from TEC's Day View) will actually be the
+		 * translated "Ongoing" text, set from
+		 * Tribe__Events__Template__Day::setup_view().
+		 *
+		 * @param $post_id
+		 *
+		 * @return string
+		 */
+		private function get_non_all_day_time_slot_name( $post_id ) {
+			$timezone = Tribe__Events__Timezones::wp_timezone_string();
+
+			$existing_timezone = date_default_timezone_get(); // will fallback to UTC but may also return a TZ environment variable (e.g. EST)
+
+			if ( ! in_array( $timezone, timezone_identifiers_list() ) ) {
+				$timezone = get_option( 'timezone_string' ); // could return NULL
+			}
+
+			if ( empty( $timezone ) ) {
+				$timezone = $existing_timezone;
+			}
+
+			date_default_timezone_set( $timezone );
+
+			$event_start_timestamp = $this->get_timestamp( $post_id );
+			$event_end_timestamp   = $this->get_timestamp( $post_id, 'End' );
+
+			$day_view_ymd = get_query_var( 'eventDate' );
+
+			$day_view_midnight_timestamp   = strtotime( sprintf( '%s 00:00:00', $day_view_ymd ) );
+			$day_view_end_of_day_timestamp = strtotime( sprintf( '%s 23:59:59', $day_view_ymd ) );
+
+			if ( $day_view_midnight_timestamp > $event_start_timestamp ) {
+				if ( $day_view_end_of_day_timestamp < $event_end_timestamp ) {
+					return $this->get_all_day_text();
+				} else {
+					$hour = - 1;
+				}
+			} else {
+				$hour = date( 'G', $event_start_timestamp );
+			}
+
+			// set back to what date_default_timezone_get() was
+			date_default_timezone_set( $existing_timezone );
+
+			foreach ( $this->get_time_of_day_ranges() as $time_of_day => $hours ) {
+				if ( in_array( $hour, $hours ) ) {
+					return (string) $time_of_day;
+				}
+			}
+		}
+
+		/**
+		 * Get the timestamp of "right now" in the timezone from WordPress settings.
+		 *
+		 * Note: `current_time( 'timestamp' )` does not work for our purposes.
+		 *
+		 * @return int
+		 */
+		public static function now_timestamp() {
+			$timezone = Tribe__Events__Timezones::wp_timezone_string();
+
+			$existing_timezone = date_default_timezone_get(); // will fallback to UTC but may also return a TZ environment variable (e.g. EST)
+
+			if ( ! in_array( $timezone, timezone_identifiers_list() ) ) {
+				$timezone = get_option( 'timezone_string' ); // could return NULL
+			}
+
+			if ( empty( $timezone ) ) {
+				$timezone = $existing_timezone;
+			}
+
+			date_default_timezone_set( $timezone );
```

Comment:
WordPress sets the default timezone to UTC and various bits of code in WP itself (and in our own plugins) rely on that being the case ... we should be extra careful about changing it. Is this needed?